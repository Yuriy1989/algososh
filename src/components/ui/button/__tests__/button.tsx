import { fireEvent } from "@testing-library/dom";
import { render, screen,  } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Button } from "../button";

const buttonText = "Развернуть";

describe("Snapshot test button component", () => {
  it("button with text", () => {
    const tree = renderer.create(<Button text={buttonText} />).toJSON()
    expect(tree).toMatchSnapshot();
  })

  it("button without text", () => {
    const tree = renderer.create(<Button />).toJSON()
    expect(tree).toMatchSnapshot();
  })

  it("button disabled", () => {
    const tree = renderer.create(<Button disabled/>).toJSON()
    expect(tree).toMatchSnapshot();
  })

  it("button with loading", () => {
    const tree = renderer.create(<Button isLoader={true}/>).toJSON()
    expect(tree).toMatchSnapshot();
  })
});

it("button with event click", () => {
  window.alert = jest.fn();

  render(<Button text="click button" onClick={() => { alert('callback') }} />)
  const button = screen.getByText("click button");
  fireEvent.click(button);
  expect(window.alert).toHaveBeenCalledWith("callback")
})
