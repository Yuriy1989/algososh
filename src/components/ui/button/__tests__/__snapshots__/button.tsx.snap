// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test button component button disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Snapshot test button component button with loading 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Snapshot test button component button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Развернуть
  </p>
</button>
`;

exports[`Snapshot test button component button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
